cobra/flux_analysis/parsimonious.py

```diff
@@ -1,22 +1,151 @@
+    the cell will try to minimize overall flux while still maximizing the
+    original objective function, e.g. the growth rate.
+
+    Parameters
+    ----------
+    model : cobra.Model
+        The model
+
+    already_irreversible : bool, optional
+        By default, the model is converted to an irreversible one.
+        However, if the model is already irreversible, this step can be
+        skipped. Ignored for optlang solvers as not relevant.
+
+    fraction_of_optimum : float, optional
+        Fraction of optimum which must be maintained. The original objective
+        reaction is constrained to be greater than maximal_value *
+        fraction_of_optimum.
+
+    desired_objective_value : float, optional
+        A desired objective value for the minimal solution that bypasses the
+        initial optimization result. Ignored for optlang solvers, instead,
+        define your objective separately and pass using the `objective`
+        argument.
+
+    objective : dict or model.solver.interface.Objective
+        A desired objective to use during optimization in addition to the
+        pFBA objective. Dictionaries (reaction as key, coefficient as value)
+        can be used for linear objectives. Not used for non-optlang solvers.
+
+    solver : str, optional
+        Name of the solver to be used. If None it will respect the solver set
+        in the model (model.solver).
+
+    **optimize_kwargs : additional arguments for legacy solver, optional
+        Additional arguments passed to the legacy solver. Ignored for
+        optlang solver (those can be configured using
+         model.solver.configuration).
+
+    Returns
+    -------
+    cobra.core.Solution.Solution
+        The solution object obtained optimizing the model with the added
+        pFBA constraints.
+
+    References
+    ----------
+    .. [1] Lewis, N. E., Hixson, K. K., Conrad, T. M., Lerman, J. A.,
+       Charusanti, P., Polpitiya, A. D., … Palsson, B. Ø. (2010). Omic data
+       from evolved E. coli are consistent with computed optimal growth from
+       genome-scale models. Molecular Systems Biology, 6,
+       390. doi:10.1038/msb.2010.47
+
+    """
+    legacy, solver = sutil.choose_solver(model, solver)
+    if not legacy:
+        return _optimize_minimal_flux_optlang(
+            model, objective=objective,
+            fraction_of_optimum=fraction_of_optimum)
+    else:
+        return _optimize_minimal_flux_legacy(
+            model, already_irreversible=already_irreversible,
+            fraction_of_optimum=fraction_of_optimum,
+            solver=solver,
+            desired_objective_value=desired_objective_value,
+            **optimize_kwargs)
+
+
+def add_pfba(model, objective=None, fraction_of_optimum=1.0):
+    if objective is not None:
+        model.objective = objective
+    model.fix_objective_as_constraint(fraction=fraction_of_optimum)
+    pfba_objective = model.solver.interface.Objective(add(
```

Comment:
This should only include variables belonging to reactions. There might be other variables defined in the solver that should not be minimized.